apps/prairielearn/src/sync/syncFromDisk.ts

```diff
@@ -29,6 +31,33 @@ interface Logger {
   verbose: (msg: string) => void;
 }
 
+export async function validateCourseData(
+  courseId: string,
+  courseDir: string,
+  logger: Logger,
+): Promise<{ courseData: courseDB.CourseData; sharingSyncError: boolean }> {
+  const courseData = await courseDB.loadFullCourse(courseId, courseDir);
+
+  if (config.checkSharingOnSync) {
+    // TODO: also check if questions were un-shared in the JSON or if any
+    // sharing sets were deleted
```

Comment:
In the context of my other comment: I acknowledge that in the near future you'll need more data than just a glorified list of directories. Even when that's the case, I think we'll be better writing off highly-efficient, possibly bespoke code here than relying on `loadFullCourse`.